lua/iswap/internal.lua

```diff
@@ -166,6 +170,30 @@ function M.swap_nodes_and_return_new_ranges(a, b, bufnr, should_move_cursor)
   return { a_data, b_data }
 end
 
+function M.move_node_to_index(children, cur_node_idx, a_idx, config)
+  local bufnr = vim.api.nvim_get_current_buf()
+  if a_idx == cur_node_idx + 1 or a_idx == cur_node_idx - 1 then
+    -- This means the node is adjacent, swap and move are equivalent
+    return M.swap_nodes_and_return_new_ranges(children[cur_node_idx], children[a_idx], bufnr, config.move_cursor)
+  end
+
+  local children_ranges = vim.tbl_map(function(node) return { node:range() } end, children)
+  local cur_range = children_ranges[cur_node_idx]
+
+  local incr = (cur_node_idx < a_idx) and 1 or -1
+  local ret_a, ret_b
+  for i = cur_node_idx + incr, a_idx, incr do
+    local a_range, b_range =
+      unpack(M.swap_ranges_and_return_new_ranges(cur_range, children_ranges[i], bufnr, config.move_cursor))
+    if not ret_a then ret_a = a_range end
+    ret_b = b_range
+    cur_range = b_range
+    children_ranges[i] = a_range
+  end
+
+  return { ret_a, ret_b }
```

Comment:
I feel like only `ret_b` needs to be `flash_confirm`'ed here, otherwise the UI makes it look like a swap, which is a bit confusing. `ret_a` is highlighted seemingly arbitrarily almost, since it wasn't exactly involved/special in the operation

Reply:
so what do you think of just removing calculatins for `ret_a`, renaming `ret_b` -> `ret`

Reply:
Can do

Reply:
Also, for some reason I can't reply directly above but basically if find_cur_node is true and cur_nodes is empty, that means the cursor happens to lie in the parent node (the list) but not in any of the children. This happens if you are on the braces of a list for example. It doesn't make sense to use this list nodes for ISwapWith, because there is no cur_node, so we just skip it. However we can still use this list for ISwap because the user will use the labels to select the two nodes.